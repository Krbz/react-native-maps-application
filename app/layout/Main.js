import React, { Component } from "react";
import { Dimensions, DrawerLayoutAndroid, Text, View } from "react-native";

import { connect } from "react-redux";
import { styles } from "../styles.android";
import MenuLayout from "./Menu";
import MapLayout from "./Map";

const { width } = Dimensions.get('window');

export class MainLayout extends Component {
  constructor(props) {
    super(props);
    this.state = {
      drawerReducer: {
        instance: null,
        visible: false,
      }
    };

    this.showDrawer = this.showDrawer.bind(this);
    this.hideDrawer = this.hideDrawer.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.drawerReducer && nextProps.drawerReducer !== this.state.drawerReducer) {
      this.setState({
        drawer: Object.assign({}, this.state.drawerReducer, nextProps.drawerReducer),
      });

      if (nextProps.drawerReducer.visible) {
        this.showDrawer();
      } else {
        this.hideDrawer();
      }
    }
  }

  showDrawer() {
    this.state.drawerReducer.instance.openDrawer();
  }

  hideDrawer() {
    this.state.drawerReducer.instance.closeDrawer();
  }

  render() {
    return (
      <View style={styles.mainLayout}>
        <DrawerLayoutAndroid
          style={styles.AppContainer}
          ref={(drawer) => this.state.drawerReducer.instance = drawer}
          drawerWidth={width - (width / 5)}
          renderNavigationView={() => <MenuLayout/>}>
          <View style={{flex: 1, alignItems: 'center'}}>
            <MapLayout />
          </View>
        </DrawerLayoutAndroid>
      </View>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return {
    drawerReducer: state.drawerReducer,
  }
}
/* istanbul ignore next */
function mapDispatchToProps(dispatch) {
  return {

  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MainLayout);
