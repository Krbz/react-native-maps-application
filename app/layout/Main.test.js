import "react-native";
import React from "react";
import { MainLayout } from "./Main.js";
import renderer from "react-test-renderer";

test('renders correctly', () => {
  const tree = renderer.create(
    <MainLayout />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});