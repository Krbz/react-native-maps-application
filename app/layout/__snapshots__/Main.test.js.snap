// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "bottom": 0,
      "justifyContent": "flex-end",
      "left": 0,
      "margin": 5,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderColor": "red",
          "borderWidth": 1,
        },
        Object {
          "backgroundColor": "#eeeeee",
        },
      ]
    }
  />
</View>
`;
